modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_sharing.jsp

```diff
@@ -21,6 +21,18 @@ String returnToFullPageURL = ParamUtil.getString(request, "returnToFullPageURL")
 
 Portlet portlet = PortletLocalServiceUtil.getPortletById(company.getCompanyId(), portletResource);
 
+boolean portletPublished = true;
+
+if (layout.isDraftLayout()) {
+	Layout publishedLayout = LayoutLocalServiceUtil.getLayout(layout.getClassPK());
+
+	LayoutTypePortlet publishedLayoutTypePortlet = (LayoutTypePortlet)publishedLayout.getLayoutType();
+
+	portletPublished = publishedLayoutTypePortlet.hasPortletId(portletResource);
+
+	themeDisplay.setLayout(publishedLayout);
+}
+
 String widgetURL = PortalUtil.getWidgetURL(portlet, themeDisplay);
 %>
```

Comment:
Hi @IstvanD, 

Please move `String widgetURL = PortalUtil.getWidgetURL(portlet, themeDisplay); ` to `<c:when test="<%= portletPublished %>">`. 

I think your previous solution way is better than the current way. I mean setting the draft layout to themeDisplay after generating these urls. This is because the tags might use layout from themeDisplay in the future during the form renderring.

So please use the previous way. Like:

```
Layout publishedLayout = null;

if (layout.isDraftLayout()) {
	publishedLayout = LayoutLocalServiceUtil.getLayout(layout.getClassPK());

	LayoutTypePortlet publishedLayoutTypePortlet = (LayoutTypePortlet)publishedLayout.getLayoutType();

	portletPublished = publishedLayoutTypePortlet.hasPortletId(portletResource);
}
%>
....
<c:when test="<%= portletPublished %>">
		<%
			themeDisplay.setLayout(publishedLayout);
			String widgetURL = PortalUtil.getWidgetURL(portlet, themeDisplay);
			String opensocialGadgetURL = PortalUtil.getGoogleGadgetURL(portlet, themeDisplay);
			String netvibesWidgetURL = PortalUtil.getNetvibesURL(portlet, themeDisplay);
			themeDisplay.setLayout(layout);
		%>
```

Reply:
hi @yuhai I made the changes you suggested with one small modification: I put the line
`themeDisplay.setLayout(publishedLayout);`
before
`<c:when test="<%= portletPublished %>">`
because if the layout is a widget page, `layout.isDraftLayout()` evaluates to **false** and `publishedLayout` will be **null**.